### ISSUE TITLE
`TypeTestUtil.isA` confuses type `Character` from same package with `java.lang.Character`

### ISSUE THREAD

**Issue by Luro02** [pmd/pmd#4690]
**Affects PMD Version:** 7.0.0-rc3 

**Description:**

`TypeTestUtil.isA` is used by [`PrimitiveWrapperInstantiationRule`](https://github.com/pmd/pmd/blob/master/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/PrimitiveWrapperInstantiationRule.java) to check if a type is a primitive like a `Character`. The implementation is not correct, because with this code:

`src/Character.java`
```java
public class Character {
    public Character(String string, Integer integer) {}
}
```

`src/Test.java`
```java
public class Test {
    public void execute() {
        Character character = new Character("a", 1);
    }
}
```

the lint reports the `new Character` line, which is wrong. The code does compile and the constructor does not exist in `java.lang.Character`.

**Comment by jsotuyod** [comment]
@Luro02 how are you running PMD? This seems to be caused simply because you are not setting the `auxclasspath` properly https://docs.pmd-code.org/latest/pmd_languages_configuration.html#java-language-properties

**Comment by Luro02** [comment]
I did not set that, seems to have been the issue
